### PR TITLE
fix(DB/CreatureScript): Deathblow to the Legion

### LINKED ISSUE #470: [Quest][Outland] Deathblow to the Legion | 10409
**Description**:

This quest is bugged due to no script support

Deathblow to the Legion
http://www.wowhead.com/quest=10409 

**Current behaviour**: 

You can kill http://www.wowhead.com/npc=20132/socrethar, but it won't credit the quest, just will credit his death, but, as there is no script, and no dialogue or event, the quest won't be completed.

image

**Expected behaviour**: 

Event should happen, i attach link to youtube video where you can see the quest on live realm.

https://www.youtube.com/watch?v=Ncfb5hyiy9E

**Steps to reproduce the problem**:

1. .quest add 10409
2. Go coordinates and use item provided by quest: http://www.wowhead.com/item=29796
.go xyz coordinates on image
image

3. .go xyz coordinates on 2nd image:
image

4. Kill Sorcrethar

5. Quest is not completed 🔪 

**Branch(es)**: master

**AC hash/commit**:  604fa387b53d3d298840026b83caebc891275849

**Operating system**:  Debian

**Modules**:  Mod-Transmog, Mod-Anticheat

**Other Customizations**: No.

**Additional data: 

There is a script at OregonCore repo, probably working, but no SQL data provided, and also needs to be adapted:

https://github.com/OregonCore/OregonCore/blob/0f4d552fce1242f49f625d9ac54f7f3aa0c349a9/src/scripts/Outland/netherstorm.cpp#L2468

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/44516730-quest-outland-deathblow-to-the-legion-10409?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Stoabrogga**
Co-authored-by: Pondaveia Pondaveia@users.noreply.github.com

##### CHANGES PROPOSED:
Implement the event which takes place during the "Deathblow to the Legion" quest. This is based on PR #1834 but had to be heavily modified as the original scripts contained many bugs and inconsistencies.

###### ISSUES ADDRESSED:
Closes #470

##### TESTS PERFORMED:
- tested build on Ubuntu 16.04 / clang 7
- tested successfully in-game

##### HOW TO TEST THE CHANGES:
Alliance quest "Deathblow to the Legion":
- preparation:
```
.quest remove 10409
.quest add 10409
.go 4773.160645 3450.580322 105.150002 530
```
- follow the quest by using "Socrethar's Teleportation Stone" and then talk to Adyen

Please also test the Horde version "Turning Point":
- preparation:
```
.quest remove 10507
.quest add 10507
.go 4928.93 3823.68 211.483 530
```
- kill Socrethar (you can use "Voren'thal's Presence" to weaken him)

##### KNOWN ISSUES AND TODO LIST:
none

##### Target branch(es):
Master

### CODE REVIEW

**Comment by Selyria** [comment]
So I tested the PR and have to say, this is a super fix.
Tested on both sides.

**Comment by Stoabrogga** [comment]
@Selyria 
Great, thanks for testing this 👍 

> Kaylaan the Lost used the wrong spell for Divine Shield the right one would be 13874

I tried this spell, but for some unknown reason it does not provide the correct visual spell effect. As this is only for a roleplay event I think it is ok to use another spell which provides an appropriate effect.

**Comment by Selyria** [comment]
> @Selyria
> Great, thanks for testing this 👍
> 
> > Kaylaan the Lost used the wrong spell for Divine Shield the right one would be 13874
> 
> I tried this spell, but for some unknown reason it does not provide the correct visual spell effect. As this is only for a roleplay event I think it is ok to use another spell which provides an appropriate effect.

yeah, it's perfectly okay. The main thing is that it works and that does it very well.

File: data/sql/updates/pending_db_world/rev_1562779174835710547.sql [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+(610210, 5, 4884.83, 3808.67, 198.972, 0.451103, 0, 0, 0, 100, 0),
+(610210, 6, 4889.64, 3811.75, 202.37, 0.568913, 0, 0, 0, 100, 0),
+(610210, 7, 4898.13, 3817.17, 208.018, 0.568913, 0, 0, 0, 100, 0),
+(610210, 8, 4907.47, 3823.15, 211.487, 0.568913, 0, 0, 0, 100, 0),
+(610210, 9, 4925.36, 3834.58, 211.494, 0.568913, 0, 0, 0, 100, 0);
+
+-- Karja waypath
+DELETE FROM `waypoint_data` WHERE `id` = 500020;
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES
+(500010, 0, 4808.67, 3776.1, 210.545, 0.185895, 0, 0, 0, 100, 0),
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1562779174835710547.sql]: (500020, 0, 4808.67, 3776.1, 210.545, 0.185895, 0, 0, 0, 100, 0),

Comment (Winfidonarleyan) [reply data/sql/updates/pending_db_world/rev_1562779174835710547.sql]: Yes, need `;` in end

Comment (Stoabrogga) [reply data/sql/updates/pending_db_world/rev_1562779174835710547.sql]: Good catch, thanks! I corrected this and pushed a new commit.

**Comment by poszer** [comment]
Thank you for PR :+1:

**Pull request merged**